### PR TITLE
Add memray profiler

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## Closes https://github.com/elastic/connectors-py/issues/140

Added memray profiler which will generate flamegraph reports in functional tests.

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally

#### Changes Requiring Extra Attention

- [x] New external service dependencies added.

### CODE REVIEW

**Review by tarekziade — COMMENTED** [review]
Looks good thanks! just need to move the requirement

File: requirements.txt [hunk 1]
```diff
@@ -10,6 +10,9 @@ guppy3==3.1.2
 cron-schedule-triggers==0.0.11
 pytz==2019.3
 
+# memory profiler
+memray==1.3.0
```
Comment (tarekziade) [inline requirements.txt:14]: should go in `test-requirements.txt`

File: connectors/tests/ftest.sh [hunk 2]
```diff
@@ -15,11 +15,13 @@ $ROOT_DIR/bin/fake-kibana --index-name search-$NAME --service-type $NAME --debug
 
 make load-data
 
-$ROOT_DIR/bin/elastic-ingest --one-sync --sync-now --debug
+$ROOT_DIR/bin/python -m memray run -f -o $NAME-sync.bin $ROOT_DIR/bin/elastic-ingest --one-sync --sync-now --debug
```
Comment (tarekziade) [inline connectors/tests/ftest.sh:18]: can we make it optional? I am thinking 

```
make ftest NAME=mysql MEMPROFILE=yes
```

and by default we don't do it
so we can also use the script in https://github.com/elastic/connectors-python/pull/180 
and allow:

```
make ftest NAME=mysql PERF_METRICS=yes
```

**Comment by moxarth-elastic** [comment]
Closing this pull request and will raise a new one with the similar required changes

**Pull request closed without merging**